### PR TITLE
Add link checking

### PR DESCRIPTION
**Pull request by henryre**
Parse links in .py files, then see if they return a 4XX code. Tested manually several times, and caught a bad link! Importantly, checked this finds bad readthedocs links

### CODE REVIEW

File: scripts/build.py [hunk 1]
```diff
@@ -14,6 +16,50 @@
 NOTEBOOKS_CONFIG_FNAME = ".notebooks"
 SCRIPTS_CONFIG_FNAME = ".scripts"
 
+# Credit to: https://gist.github.com/pchc2005/b5f13e136a9c9bb2984e5b92802fc7c9
+# Original source: https://gist.github.com/dperini/729294
+MARKDOWN_URL_REGEX = re.compile(
```
Comment (vincentschen) [inline scripts/build.py]: 🚀

**Comment by henryre** [comment]
negligible compared to running tutorials...but will double check travis time!

**Comment by brahmaneya** [comment]
Damn, I was about to send a PR for this

**Comment by henryre** [comment]
@brahmaneya lol sorry, feel free to improve based on what you did!!

File: spam/01_spam_tutorial.py [hunk 2]
```diff
@@ -863,7 +863,7 @@ def plot_probabilities_histogram(Y):
 # We'll use Keras, a popular high-level API for building models in TensorFlow, to build a simple logistic regression classifier.
 # We compile it with a `categorical_crossentropy` loss so that it can handle probabilistic labels instead of integer labels.
 # Using a _noise-aware loss_ &mdash; one that uses probabilistic labels &mdash; for our discriminative model lets
-# us take full advantage of the label model's learning procedure (see our [NeurIPS 2016 paper](https://arxiv.org/abs/1605.07723)).
+# us take full advantage of the label model's learning procedure (see our [NeurIPS 2016 paper](https://arxiv.org/abs/1605.07723).
```
Comment (brahmaneya) [inline spam/01_spam_tutorial.py]: Shouldn't we have a closing bracket for the (see our.. ?

Comment (henryre) [reply spam/01_spam_tutorial.py]: ohhhhhh whoops...my regexing failed...

**Pull request merged**